app/src/hooks/useSessionStorage.ts

```diff
@@ -0,0 +1,56 @@
+import { useState } from 'react'
+
+import { SessionData } from '@src/types'
+
+// Custom hook to persist state across page refresh
+export default function useSessionStorage(
```

Comment:
I like this custom hook pattern and wonder if this app would be well served by something similar for form state!